src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -460,4 +467,16 @@ public int compare(T o1, T o2) {
     public void setScheduler(RunnerScheduler scheduler) {
         this.scheduler = scheduler;
     }
+
+    private static class ClassRuleCollector implements MemberValueConsumer<TestRule> {
+        final List<TestRule> result = new ArrayList<TestRule>();
+
+        public void accept(FrameworkMember member, TestRule value) {
+            ClassRule rule = member.getAnnotation(ClassRule.class);
+            if (rule != null) {
+                RuleContainer.recordRuleOrder(value, rule.order());
```

Comment:
It's not obvious here that we are modifying a thread local. I would prefer that `recordRuleOrder()` be a non-static method.

You can use [`ThreadLocal.inititalValue()`](https://docs.oracle.com/javase/7/docs/api/java/lang/ThreadLocal.html#initialValue()) to initialize the thread local to a no-op value